### ISSUE TITLE
Left sidebar can be overwhelming and doesn't provide sufficient user value

### ISSUE THREAD

**Issue by InMatrix** [dart-lang/dartdoc#3437]
Left sidebar on a class page (e.g., [AppBar](https://api.flutter.dev/flutter/material/AppBar-class.html)) currently shows every symbols in its library in the alphabetical order. For some large libraries such as `material` in Flutter, this can be overwhelming to users, as shown in a recent [user study](https://docs.google.com/presentation/d/1Q5rY7ogq_yguICKLKiDgxpkCv74OPJivyknOxn7dhBQ/edit#slide=id.g24a771b330d_0_947).

The team discussed a few potential improvements:
- Collapsing the list by headers (e.g., Classes, CONSTANTS, ENUMS, etc). 
- Formalize the "See also" section in many Flutter docs and list the linked pages at the top of the sidebar
<img width="791" alt="image" src="https://github.com/dart-lang/dartdoc/assets/348942/06c60d97-701e-4f4b-ab07-94e396b6e59d">

- Show N most frequently used classes at the top of the sidebar. We could use Google Analytics data to identify those classes or look into the pub packages corpus.

**Comment by Levi-Lesches** [comment]
I agree here, it is way too much noise IMO in most libraries. Moreover, when viewing a class, I am way more interested in the members of that class than other classes in the library.

#3464 proposes to replace the left sidebar with the right sidebar -- ie, when viewing a class, the left sidebar lists the class's members. That way, when viewing a constructor/property/method, the sidebar doesn't "jump" from the right to the left.
